source/github-helpers/github-url.ts

```diff
@@ -16,7 +16,16 @@ export default class GitHubURL extends URL {
 		this.pathname = super.pathname;
 	}
 
-	disambiguateReference(ambiguousReference: string[]): {branch: string; filePath: string} {
+	assign(replacements: Partial<GitHubURL>): this {
+		Object.assign(this, replacements);
+		return this;
+	}
```

Comment:
It looks like this works too, it lets us assign multiple times if necessary, e.g. url.assign(opts, {route: 'blob'})
```suggestion
	assign = Object.assign.bind(this);
```

But it needs to be moved above the constructor.

Reply:
it will affect the way assign works.
I will commit so you can see